src/utils/hibp.js

```diff
@@ -189,6 +189,8 @@ async function downloadBreachIcons (breaches) {
             resolve([breachDomain, `/images/logo_cache/${breachDomain.toLowerCase()}.ico`])
           })
           file.on('error', (error) => reject(error))
+        }).on('error', (_error) => {
+          resolve(null)
```

Comment:
Do we want to log the error for 404 responses as well? Also, note that this error can occur for _every_ icon we attempt to fetch, so that's potentially a lot of logs.

(And: any guidelines on how we log? Or do I just come up with some error message and log it using a mozlog warning?)

Reply:
hmm, I'd log with mozlog warning, I wouldn't use a `error` level. I don't think we have a guideline, but good question... Does Relay have centralized logging/messaging? When @rhelmer comes back, we can maybe have a team discussion about this

Reply:
@Vinnl in that case, we can probably ignore this comment and not log
@pdehaan Thanks for the insight! With monitor refresh, we aren't actually pulling breaches using `HIBP_RELOAD_BREACHES_TIMER` anymore.. This env var will be deprecated. We relies on HIBP to push updates to us instead.

Reply:
There's two different error conditions: negative responses by DDG (e.g. a 404), or request errors (e.g. the network being down). The former, e.g. when DDG doesn't have a logo, was already dealt with (without logging). This PR only adds error handling for the latter, so it usually shouldn't lead to so many errors all the time - although _if_ the network is down, it will result in a log entry for every company.

PS. @pdehaan I do appreciate more input, especially since so many of us aren't yet intimately familiar with the project, so no need to apologise :)

Reply:
I agree this is probably fine for now (especially given how many transient and non-transient errors it sounds like we're likely to have, thanks for looking into that @pdehaan). I think once we start looking at putting the parts of Monitor that interface with external services behind queues we'll have an opportunity to have some re-usable caching and back-off strategies around failures that we can share.